src/pages/Admin/BackgroundSelect/BackgroundSelect.tsx

```diff
@@ -1,75 +1,82 @@
+
+import { FileButton } from "components/atoms/FileButton";
 
-// Typings
-import { BackgroundSelectProps } from "./BackgroundSelect.types";
+import "./BackgroundSelect.scss";
 
-// Styles
-import * as S from "./BackgroundSelect.styles";
-import { useFetchAssetImages } from "hooks/useFetchAssetImages";
+export interface BackgroundSelectProps {
+  venueName: string;
+  mapBackground?: string;
+}
 
-const BackgroundSelect: React.FC<BackgroundSelectProps> = ({
+export const BackgroundSelect: React.FC<BackgroundSelectProps> = ({
   venueName,
   mapBackground,
 }) => {
   const { user } = useUser();
-  const mapBackgrounds = useFetchAssetImages("mapBackgrounds");
+  const {
+    assets: mapBackgrounds,
+    isLoading: isLoadingBackgrounds,
+  } = useFetchAssets("mapBackgrounds");
 
-  const handleUpload = (url: string, file?: FileList) => {
-    if (!user) return;
+  const [{ loading: isUploading }, uploadMapBackground] = useAsyncFn(
+    async (url: string, file?: FileList) => {
+      if (!user) return;
 
-    if (file && file.length) {
-      return updateVenue_v2(
+      const hasCustomBackground = !!(file && file.length);
+
+      return await updateVenue_v2(
         {
           name: venueName,
-          mapBackgroundImageFile: file,
+          ...(hasCustomBackground && { mapBackgroundImageFile: file }),
           mapBackgroundImageUrl: url,
         },
         user
       );
-    }
+    },
+    [user, venueName]
+  );
 
-    return updateVenue_v2(
-      {
-        name: venueName,
-        mapBackgroundImageUrl: url,
-      },
-      user
-    );
-  };
+  const hasBackgrounds = !!mapBackgrounds.length && !isLoadingBackgrounds;
 
   return (
-    <S.Wrapper hasImage={!!mapBackground}>
+    <div className="BackgroundSelect">
       {!mapBackground && (
         <>
-          <FileButton onChange={handleUpload} />
+          <FileButton
+            disabled={isUploading}
+            title="Import a map background"
+            description="Recommended size: 2000px / 1200px"
+            onChange={uploadMapBackground}
+          />
 
           <h3>Or choose a map</h3>
-          <S.MapBrowserGrid>
-            {mapBackgrounds.length > 0 &&
-              mapBackgrounds.map((item, index) => (
-                <S.MapItem
-                  backgroundImage={item}
+          {isLoadingBackgrounds && <div>Loading maps...</div>}
+
+          <div className="BackgroundSelect__map-grid">
+            {hasBackgrounds &&
+              mapBackgrounds.map((mapBackground, index) => (
+                <button
+                  className="BackgroundSelect__map"
+                  disabled={isUploading}
+                  style={{ backgroundImage: `url(${mapBackground})` }}
                   key={index}
-                  aspectRatio="1.6/1"
-                  onClick={() => handleUpload(item)}
+                  onClick={() => uploadMapBackground(mapBackground)}
                 />
               ))}
-          </S.MapBrowserGrid>
+          </div>
```

Comment:
useMemo

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1583/commits/f8fa0c67c34ac626919737b4bbd112b4bd715553